### ISSUE TITLE
AdaptiveCard schema field_name not set correctly

### ISSUE THREAD

**Issue by jwrackware** [dennis6p/adaptive-cards-py#24]
The schema field name for the AdapterCard is not being set correctly. Example:

```
>>> from adaptive_cards.card import AdaptiveCardBuilder
>>> builder = AdaptiveCardBuilder()
>>> card = builder.create()
>>> card.to_json()
'{"type": "AdaptiveCard", "version": "1.0", "schema": "http://adaptivecards.io/schemas/adaptive-card.json"}'
```

As I understand it, the schema field name should be `$schema`. I believe the problem is that `dataclass_json.config` sets up the `letter_case` function to update the field name but the `field_name` argument is not being passed to `dataclass_json.config`. Instead, it is added to the metadata dictionary after the fact. Hence, the `letter_case` function will not change the field name.

I originally noticed this when trying to debug a card that was failing to render in Microsoft Teams (seems like a size limit problem at this point). I thought maybe the schema field name was part of the problem but I implemented a workaround that changed the field name and that didn't help with my issue. So, it may be that this issue doesn't cause any problems, but it still wouldn't hurt to fix it (assuming it actually is an issue and I'm not simply mistaken).

**Comment by dennis6p** [comment]
Nice catch. And thanks for providing me part of the solution already. For some reason, I did not define the field name as part of the config, but in the dict to be concatenated instead. I fixed this and an exported object should now contain the `$schema` field properly named. Your observation also seems to be true. The missing `$` is not necessarily an issue when it comes to sending and rendering cards in MS Teams. However, worth to be fixed anyway. I addressed your point in #25.

Indeed, there seems to be a limit when it comes to the total card size. I don't know if they changed the limit in the meantime, but as I was running into this issue some time ago, the maximum size was still 28kb. Here are some further readings on that topic:

https://answers.microsoft.com/en-us/msteams/forum/all/entity-size-too-large-error-when-the-adaptive-card/b5e99337-0df5-41e1-804e-1ef2a7e3b0e5

https://github.com/microsoft/BotFramework-Services/issues/228

https://techcommunity.microsoft.com/discussions/teamsdeveloper/bot-message-limitation-and-alternative-for-that-limitation/3941571

Thanks for your contribution!

**Comment by jwrackware** [comment]
That was quick. Awesome work.

Regarding the size limit, do you remember how you were measuring the size? And if you found you were able to send cards all the way up to that 28k max size? I've been measuring the number of characters in the JSON string sent to the Webhook but I can only get a little over 11,000 characters before I start to see the 413 HTTP response. It might be an encoding issue when using requests, but I'm still looking into that.

**Comment by dennis6p** [comment]
Thanks. I appreciate that!

Hm, good question. It's some time ago already, when I faced this issue. But maybe this size check could be integrated into the `SchemaValidator`, as this can be considered as a hard requirement from the target framework (e.g. MS Teams). 

But there are some small tricks, how one could reduce the overall size by optimizing the cards structure. Maybe ChatGPT has some hints about that in the meantime.

**Comment by jwrackware** [comment]
So, I built a minimal card with just a TextBlock with a long string in it. With that setup, I was able to get a JSON string with 21291 characters to render in Teams (21292 will not render). Then I shortened the string and added a second TextBlock 
and now I can only render up to 21183 JSON characters. Very strange. Probably just another Microsoft product with lots of quirks.

Anyway, this is not your problem to solve. Thanks again for fixing the schema field name thing.
